### PR TITLE
Fix #2695

### LINKED ISSUE #2695: `:tabopen` broken in FF80
Tested with fresh profiles.

## FF79

Works as expected.

## FF80b2

```
:tabopen bla bla
```
works

```
:tabopen search bla bla
```
works

```
:tabopen google bla bla
```
broken (default new tab window)

```
:tabopen https://google.com
```
broken (default new tab window)

---

`:open` does not suffer from the same issue.

### PR DESCRIPTION
**Pull request by bovine3dom**
Annoyingly, I can reproduce #2695 with a real profile but not via ` yarn run build && eval (yarn bin)/web-ext run -s build/ -u txti.es --firefox=(which firefox-developer-edition)`.

(Edit: and when I do reproduce it on a real profile I don't see any errors in the web or developer consoles).

### CODE REVIEW

**Comment by bovine3dom** [comment]
I should temporarily reduce the number of retries now that the tests are really broken.

**Comment by bovine3dom** [comment]
My money is on ea257f3644f3448200239a71d2b6c4ca76ab11ae being the culprit. I'll try to wrap my head around that and avoid the redirect via the new tab page (it looks pretty janky anyway).

**Comment by bovine3dom** [comment]
@bovine3dom don't merge this without reverting the "tests can fail quicker" commit.

File: src/excmds.ts [hunk 1]
```diff
@@ -2212,18 +2212,36 @@ export async function tabopen(...addressarr: string[]) {
...
+    const args = { active } as any
+    // Ensure -c has priority.
+    if (container) {
+        if (container !== "firefox-default") {
+            args.cookieStoreId = container
+        }
+    } else if (containerId && config.get("tabopencontaineraware") === "true") {
+        args.cookieStoreId = containerId
+    }
+    // return openInNewTab(query.join(" "), args).then(tab => openInTab(tab, { loadReplace: true }, query))
```
Comment (bovine3dom) [inline src/excmds.ts:2225]:
```suggestion
```

**Comment by glacambre** [comment]
@bovine3dom I don't understand what was wrong and why your changes fix the problem. Could you explain ? (Just curious, feel free not to reply if you don't have the time :))

File: src/excmds.ts [hunk 2]
```diff
@@ -2212,18 +2212,35 @@ export async function tabopen(...addressarr: string[]) {
...
+        if (await firefoxVersionAtLeast(80)) {
+            // work around #2695 until we can work out what is going on
+            if (args.active === false || args.cookieStoreId !== undefined) {
+                throw new Error("Firefox search engines do not support containers or background tabs in FF >80. `:set searchengine google` or see issue https://github.com/tridactyl/tridactyl/issues/2695")
             }
-        } else if (containerId && config.get("tabopencontaineraware") === "true") {
-            args.cookieStoreId = containerId
+
+            // This ignores :set tabopenpos / issue #342. TODO: fix that somehow.
+            return browser.search.search(maybeURL)
```
Comment (bovine3dom) [inline src/excmds.ts:2238]: @glacambre at the heart of it it's quite a simple change: I don't open a new tab and then loadreplace it, instead we open a new tab directly.

I don't know why your method stopped working reliably in FF80. I don't really know how I knew it was your method that was the problem either - it was just a hunch as it was the most convoluted thing I could find to do with tabopen. I thought it looked a bit janky in use so I thought it was a good thing to remove anyway :)

**Pull request merged**